src/gatsby/plugins/gatsby-remark-variables/index.js

```diff
@@ -57,9 +57,8 @@ module.exports = async ({markdownAST, markdownNode}, options) => {
         try {
           result = scopedEval(expr, {...scope, page});
         } catch (err) {
-          // no-op. We previously had a warning here, but the we have so many
-          // "variable-like" constructs in our codeblocks that this becomes
-          // very spammy.
+          console.error(`Failed to interpolate expression: "${expr}"`);
+          throw err;
```

Comment:
This means builds will fail if you try and use variable interpolation without it able to resolve the variable.